### DISCUSSION TITLE
Support for Google Spanner

### DISCUSSION THREAD

**Discussion by canberksinangil** [aarondl/sqlboiler#1198] [Ideas]
Is there any plan to support [Google Spanner](https://cloud.google.com/spanner)?

As is mentioned:

> Note: SQLBoiler supports out of band driver support so you can make your own

What is the most effective way if I want to contribute to that? Is there any guidance that you can provide me?

Thanks

**Comment by canberksinangil** [comment]
https://github.com/googleapis/go-sql-spanner

**Comment by stephenafamo** [comment]
The primary thing is to be able to retrieve table and column information for code generation. [The important thing is to implement this interface](https://github.com/volatiletech/sqlboiler/blob/v4.13.0/drivers/interface.go#L36).

As an example, SQLite's driver is [here](https://github.com/volatiletech/sqlboiler/blob/master/drivers/sqlboiler-sqlite3).

Some extra modifications may be needed for the query object, perhaps to support some Spanner specific features. This part is currently tricky since SQLBoiler uses the same query object across all dialects.
